fastly/ngwaf/v1/workspaces/api_test.go

```diff
@@ -0,0 +1,209 @@
+	const wsMode = "log"
+	const wsIPAnonymization = "hashed"
+
+	wsAttackSignalThresholds := new(AttackSignalThresholdsInput)
+	wsAttackSignalThresholds.OneMinute = fastly.ToPointer(10000)
+	wsAttackSignalThresholds.TenMinutes = fastly.ToPointer(10000)
+	wsAttackSignalThresholds.OneHour = fastly.ToPointer(10000)
+	wsAttackSignalThresholds.Immediate = fastly.ToPointer(true)
+
+	var wss *Workspaces
+	var err error
+
+	// List all workspaces.
+	fastly.Record(t, "list_workspaces", func(c *fastly.Client) {
+		wss, err = List(c, &ListInput{})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Make sure the test workspace we're going to create isn't among them.
+	for _, ws := range wss.Data {
+		if ws.Name == wsName {
+			t.Errorf("found test workspace %q, aborting", wsName)
+		}
+	}
+
+	// Create a test workspace.
+	var ws *Workspace
+	fastly.Record(t, "create_workspace", func(c *fastly.Client) {
+		ws, err = Create(c, &CreateInput{
+			Name:                   fastly.ToPointer(wsName),
+			Description:            fastly.ToPointer(wsDescription),
+			Mode:                   fastly.ToPointer(wsMode),
+			IPAnonymization:        fastly.ToPointer(wsIPAnonymization),
+			AttackSignalThresholds: wsAttackSignalThresholds,
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if ws.Name != wsName {
+		t.Errorf("unexpected workspace name: got %q, expected %q", ws.Name, wsName)
+	}
+	if ws.Description != wsDescription {
+		t.Errorf("unexpected workspace description: got %q, expected %q", ws.Description, wsDescription)
+	}
+	if ws.Mode != wsMode {
+		t.Errorf("unexpected workspace mode: got %q, expected %q", ws.Mode, wsMode)
+	}
+	if ws.IPAnonymization != wsIPAnonymization {
+		t.Errorf("unexpected workspace IP anonymization: got %q, expected %q", ws.IPAnonymization, wsIPAnonymization)
+	}
+	// TODO: Update the following assertions for the AttackSignalThresholds settings once CDTOOL-1072 has been resolved and deployed.
+	// Note: At present, the API does not honor the AttackSignalThresholds settings when making a POST request to /ngwaf/v1/workspaces.
+	if ws.AttackSignalThresholds.Immediate != false {
+		t.Errorf("unexpected workspace attack signal thresholds immediate parameter: got %t, expected %t", ws.AttackSignalThresholds.Immediate, *wsAttackSignalThresholds.Immediate)
+	}
+	if ws.AttackSignalThresholds.OneMinute != 0 {
+		t.Errorf("unexpected workspace attack signal thresholds one_minute parameter: got %v, expected %v", ws.AttackSignalThresholds.OneMinute, *wsAttackSignalThresholds.OneMinute)
+	}
+	if ws.AttackSignalThresholds.TenMinutes != 0 {
+		t.Errorf("unexpected workspace attack signal thresholds ten_minutes parameter: got %v, expected %v", ws.AttackSignalThresholds.TenMinutes, *wsAttackSignalThresholds.TenMinutes)
+	}
+	if ws.AttackSignalThresholds.OneHour != 0 {
+		t.Errorf("unexpected workspace attack signal thresholds one_hour parameter: got %v, expected %v", ws.AttackSignalThresholds.OneHour, *wsAttackSignalThresholds.OneHour)
+	}
```

Comment:
If the `attack_signal_thresholds` settings are empty `go` assigns the zero values.